### PR TITLE
Upgrade pytest to latest version

### PR DESCRIPTION
**Pull request by chagui**
## Description
This pull-request propose to upgrade pytest to latest version (6.2.4) in order to:
- Replace the [deprecated yield fixture function decorator](https://docs.pytest.org/en/latest/deprecations.html#the-yield-fixture-function-decorator), which shows as a warning when [running tests](https://github.com/dbcli/pgspecial/runs/2548249091#step:6:22)
- Move pytest configuration to [pyproject.toml ](https://docs.pytest.org/en/latest/reference/customize.html#pyproject-toml)

I know some people are not fan of moving configuration to pyproject.toml and I would perfectly understand if you want to keep as is.

## Checklist
- [ ] I've added this contribution to the `changelog.rst`.
- [x] I installed pre-commit hooks (`pip install pre-commit && pre-commit install`), and ran `black` on my code.
- [x] Please squash merge this pull request (uncheck if you'd like us to merge as multiple commits)

### CODE REVIEW

**Comment by j-bennet** [comment]
What is the benefit of moving pytest settings into pyproject.toml?

File: pyproject.toml [hunk 1]
```diff
@@ -20,3 +20,9 @@ exclude = '''
 )/
 '''
 
+[tool.pytest.ini_options]
```
Comment (j-bennet) [inline pyproject.toml:23]: According to this, pytest.ini will be preferred over all other config files:
https://docs.pytest.org/en/latest/reference/customize.html#pytest-ini

And we do have [pytest.ini](https://github.com/dbcli/pgspecial/blob/master/tests/pytest.ini). If you want to move the settings, you'd have to delete that.

**Review by j-bennet — COMMENTED** [review]
Bumping pytest is great. I'm curious to hear your reasons for preferring pyproject.toml over pytest.ini though.

File: pyproject.toml (hunk 1 above)
Comment (chagui) [reply pyproject.toml:23]: I believe I did: https://github.com/dbcli/pgspecial/pull/111/files?file-filters%5B%5D=.ini :innocent:

**Comment by chagui** [comment]
The benefit would be less configuration files in the project, and one place to lookup different configuration (project one or tools). However the `pyproject.toml` is young and not adopted by every tool yet, and I know some project like it some other not so I am totally fine removing that part from the PR. It was more a suggestion rather than "we should do it this way"

File: pyproject.toml (hunk 1 above)
Comment (j-bennet) [reply pyproject.toml:23]: Hah. :) Great. I must have missed that.

**Comment by j-bennet** [comment]
> The benefit would be less configuration files in the project, and one place to lookup different configuration (project one or tools). However the pyproject.toml is young and not adopted by every tool yet, and I know some project like it some other not so I am totally fine removing that part from the PR. It was more a suggestion rather than "we should do it this way"

I see. I don't have a strong preference for pytest.ini over pyproject.toml. If this gets us one less configuretion file, and does not break anything, sure, why not.

I think if we do this, we should mention in [contributor guide](https://github.com/dbcli/pgspecial/blob/master/DEVELOP.rst) where we configure pytest, otherwise people might not see the config and add pytest.ini back.

**Review by j-bennet — CHANGES_REQUESTED** [review]
Please update the [contributor guide](https://github.com/dbcli/pgspecial/blob/master/DEVELOP.rst) with instructions on where pytest settings should go. Then it's good to merge. Thank you!

File: DEVELOP.rst [hunk 2]
```diff
@@ -117,9 +117,10 @@ After that, tests can be run with:
 
 ::
 
-    $ cd tests
```
Comment (chagui) [inline DEVELOP.rst:120]: Due to [testpaths](https://docs.pytest.org/en/6.2.x/reference.html#confval-testpaths) being added

**Review by j-bennet — APPROVED** [review]
Looks good! Thank you for the PR! 🍓

**Pull request merged**